posts/2023-05-01-wasi-target-rename-1.71.0.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Renaming the WASI target in Rust 1.71.0"
+author: Yoshua Wuyts
+---
+In Rust 1.71 the existing `wasm32-wasi` target will be renamed
+to `wasm32-wasi-preview1`. This matches the naming introduced in the [WASI
+Preview Framework][wpf], whose scheme has already been adopted by other
+language toolchains such as [Golang] and [Node.js]. It is expected that
+eventually WASI will eventually stabilize far enough that it will reach a 1.0
+status. By freeing up the `wasm32-wasi` target name now, it enables us to
+reintroduce it in the future as the target name for an eventual 1.0 stable WASI
```

Comment:
I hesitate to use the term stable here since these targets are also stable they just hint that they will be superseded by a future release. Perhaps instead of call the eventual 1.0 release "stable" we can say something akin to:

> By freeing up the `wasm32-wasi` target name now, it enables us to
reintroduce it in the future as the target name for an eventual 1.0 WASI release where evolution of WASI is expected to slow considerably in pace. All wasi targets are in and of themselves stable, but it's important to note that future wasi releases are expected to supersede previous versions which will not receive feature updates.